hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/protocol/ScmBlockLocationProtocolServerSideTranslatorPB.java

```diff
@@ -94,9 +95,33 @@ public ScmBlockLocationProtocolServerSideTranslatorPB(
         .setTraceID(traceID);
   }
 
+  private boolean isLeader() throws ServiceException {
+    if (!(impl instanceof SCMBlockProtocolServer)) {
+      throw new ServiceException("Should be SCMBlockProtocolServer");
```

Comment:
nit: might be useful to print which class current `impl` is.

Reply:
Updated